api/types/kubernetes.go

```diff
@@ -344,6 +348,16 @@ func (k *KubernetesClusterV3) Copy() KubeCluster {
 	return utils.CloneProtoMsg(k)
 }
 
+// GetStatus gets the kube cluster status.
+func (k *KubernetesClusterV3) GetStatus() KubernetesClusterStatus {
+	return k.Status
+}
+
+// SetStatus sets the kube cluster status.
+func (k *KubernetesClusterV3) SetStatus(status KubernetesClusterStatus) {
+	k.Status = status
+}
```

Comment:
Protobuf messages should not be copied by value, it's incorrect even though there's no way to lint against it for messages defined by the gogoproto codegen.

```suggestion
// GetStatus gets the kube cluster status.
func (k *KubernetesClusterV3) GetStatus() *KubernetesClusterStatus {
	return k.Status
}

// SetStatus sets the kube cluster status.
func (k *KubernetesClusterV3) SetStatus(status *KubernetesClusterStatus) {
	k.Status = status
}
```